code/modules/mob/living/living_status_procs.dm

```diff
@@ -730,6 +730,7 @@ STATUS EFFECTS
 		return
 	var/datum/status_effect/incapacitating/floored/F = IsKnockedDown()
 	if(amount <= 0)
+		SEND_SIGNAL(src, COMSIG_KNOCKDOWN_REMOVE) // we want to cancel delayed baton knockdowns as well
```

Comment:
I feel like this should be part of the status_effect itself, in the `on_remove`

Reply:
How do you mean?

Reply:
`/datum/status_effect/incapacitating/floored` itself should send this signal to the owner it has in the `on_remove` proc. 
This would however not catch the case where this proc gets called when the user is not knocked down yet. If you want to catch that as well, then you should leave it here but also add it to the other KnockDown procs